### PR TITLE
LF-4349 Finish crowdin setup to support the addition of new languages

### PR DESCRIPTION
**Pull request by kathyavini**
**Description**

As just discussed a few minutes ago 🙂
- Add the {{ matrix.locales }} to each PR title to disambiguate them
- Make upload translations and auto-approve uploaded ~GitHub variables~ [dispatch inputs](https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputs) (with fallback to false)  so they can be changed without having to merge new versions of these files

Testing plan:
- variables have been set on the repository level
- this is opened against integration; once merged I will test toggling the environmental variables and running the upload to make sure it behaves as expected
- if successful I will open a new PR to merge to main

Jira link: https://lite-farm.atlassian.net/browse/LF-4349

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Passes test case
- [ ] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [x] Other (please explain)

Can't test; actions workflow 🙁 

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

File: .github/workflows/crowdin-upload.yml [hunk 1]
```diff
@@ -45,12 +45,12 @@ jobs:
       - name: Crowdin sync
         uses: crowdin/github-action@v2
         with:
-          upload_sources: true
+          upload_sources: ${{ vars.CROWDIN_UPLOAD_SOURCES || false }}
```
Comment (Duncan-Brain) [inline .github/workflows/crowdin-upload.yml:48]: We always want upload sources `true`.

I think you meant to add this to `upload_translations`.

Is vars enough? I was expecting something like this: 

[Github Actions Inputs](https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#example-of-onworkflow_dispatchinputs)

Comment (kathyavini) [reply .github/workflows/crowdin-upload.yml]: Ack! Thanks!

Variables should be enough. Dispatch inputs looks fancier but a little harder to set up 🤔 Is it super important they are set on the dispatch? I don't think we'll be toggling very often.

Comment (kathyavini) [reply .github/workflows/crowdin-upload.yml]: @Duncan-Brain it looks really promising

Screenshot 2024-07-29 at 11 18 16 AM

... but I'm a little bit worried about this wording in your [linked docs here](https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatch):

> This trigger only receives events when the workflow file is on the default branch.

Comment (Duncan-Brain) [reply .github/workflows/crowdin-upload.yml]: I just had never heard of variables context and was trying to figure where they came from.... I see them now in the secrets area! Interesting..  I am good with it! 

Pros for workflow inputs:
- Github does mention Variables are in beta and can change at any time.
- Inputs are workflow specific and accessible without navigating through github settings to update/revert
- Variable seem best suited for across multiple actions files

Cons for workflow inputs
- might be easier for someone else to make a mistake

Comment (Duncan-Brain) [reply .github/workflows/crowdin-upload.yml]: Oh you beat me to it, yeah that is a concerning line ! -- defaults to false so risk should be low.

Totally up to you!

**Comment by Duncan-Brain** [comment]
Question: should we also remove the auto trigger for integration branch upload?

File: .github/workflows/crowdin-upload.yml (hunk 1 above)
Comment (kathyavini) [reply .github/workflows/crowdin-upload.yml]: Despite the warning, I just ran it on the unmerged feature branch with "upload translations" checked and it seemed to upload them. Dunno if that's a good thing or not 😂

I'm interested in this option; we can always have variables in reserve as our fallback if it doesn't work, they are so easy to use.

**Comment by kathyavini** [comment]
@Duncan-Brain I'm totally happy to remove the auto-trigger! It sounds like we won't return to integration-based development for a long time and the manual trigger has been fine.

**Review by Duncan-Brain — APPROVED** [review]
We will want a PR to add these changes on main too right for it to work there? (I assume thats what the workflow inputs thing meant)

**Comment by kathyavini** [comment]
> We will want a PR to add these changes on main too right for it to work there? (I assume thats what the workflow inputs thing meant)

Yup, I was thinking to clear the CrowdIn sources and do more testing while it's on integration, and if checks out, open PR against main (just to keep down the number of commits added to main).

**Review by SayakaOno — APPROVED** [review]

**Pull request merged**